src/plugin.js

```diff
@@ -37,6 +37,12 @@ function resolveOptions(chart, options) {
 	}
 	var props = chart.$zoom;
 	options = props._options = helpers.merge({}, [options, deprecatedOptions]);
+	if (typeof options.pan.mode === 'function') {
```

Comment:
Ah, this is a bit different than what I was suggesting. I see that you're doing it because you want to call the function as the user zooms or pans and not at `update` time. I missed that as a requirement you had originally. I'm not sure it makes sense to do in `resolveOptions` given that. Maybe a better place to do it would be when setting zoomMode:

https://github.com/chartjs/chartjs-plugin-zoom/blob/b5666c46e109119d1875fe3fc59f3b91bf6bf6e3/src/plugin.js#L209

    var zoomMode = zoomOptions.mode === 'function' ? zoomOptions.mode({chart: chartInstance}) : zoomOptions.mode;